Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPISidebarCocoa.mm

```diff
@@ -32,55 +32,342 @@
+        completionHandler(makeUnexpected(sidebar.error()));
+        return;
+    }
 
+    completionHandler(sidebar.value()->title());
 }
 
 void WebExtensionContext::sidebarSetTitle(const std::optional<WebExtensionWindowIdentifier> windowIdentifier, const std::optional<WebExtensionTabIdentifier> tabIdentifier, const std::optional<String>& title, CompletionHandler<void(Expected<void, WebExtensionError>&&)>&& completionHandler)
 {
-
+    auto sidebar = getOrCreateSidebarWithIdentifiers(windowIdentifier, tabIdentifier, *this);
+    if (!sidebar) {
+        completionHandler(makeUnexpected(sidebar.error()));
+        return;
+    }
+    completionHandler({ });
 }
 
 void WebExtensionContext::sidebarGetOptions(const std::optional<WebExtensionWindowIdentifier> windowIdentifier, const std::optional<WebExtensionTabIdentifier> tabIdentifier, CompletionHandler<void(Expected<WebExtensionSidebarParameters, WebExtensionError>&&)>&& completionHandler)
 {
+    auto maybeSidebar = getOrCreateSidebarWithIdentifiers(windowIdentifier, tabIdentifier, *this);
+    if (!maybeSidebar) {
+        completionHandler(makeUnexpected(maybeSidebar.error()));
+        return;
+    }
 
+    auto& sidebar = maybeSidebar.value().get();
+    completionHandler(WebExtensionSidebarParameters {
+        .enabled       = sidebar.isEnabled(),
+        .panelPath     = sidebar.sidebarPath(),
+        .tabIdentifier = sidebar.tab().transform([](auto tab) { return tab.get().identifier(); }),
+    });
 }
 
 void WebExtensionContext::sidebarSetOptions(const std::optional<WebExtensionWindowIdentifier> windowIdentifier, const std::optional<WebExtensionTabIdentifier> tabIdentifier, const std::optional<String>& panelSourcePath, const std::optional<bool> enabled, CompletionHandler<void(Expected<void, WebExtensionError>&&)>&& completionHandler)
 {
+    auto maybeSidebar = getOrCreateSidebarWithIdentifiers(windowIdentifier, tabIdentifier, *this);
+    if (!maybeSidebar) {
+        completionHandler(makeUnexpected(maybeSidebar.error()));
+        return;
+    }
 
+    auto& sidebar = maybeSidebar.value().get();
+    sidebar.setSidebarPath(panelSourcePath);
+    // according to the sidePanel docs, `enabled` is optional with default value `true`
+    // we only need to be concerned with chrome's semantics here since sidebarAction does not have any concept of enablement
+    // see: https://developer.chrome.com/docs/extensions/reference/api/sidePanel#type-PanelOptions
+    sidebar.setEnabled(enabled.value_or(true));
+    completionHandler({ });
 }
 
 bool WebExtensionContext::isSidebarMessageAllowed()
 {
+    if (auto *controller = extensionController()) {
+        WKPreferences *preferences = controller->configuration().webViewConfiguration().preferences;
+
+        for (_WKFeature *feature in [WKPreferences _features]) {
+            if ([feature.key isEqualToString:@"WebExtensionSidebarEnabled"])
+                return [preferences _isEnabledForFeature:feature];
+        }
+    }
```

Comment:
This would be good as a helper method on `WebExtensionController` that takes a string. We will likely want this for other features in the future.

```cpp
bool isFeatureEnabled(const String& featureName) const;
```